repose-aggregator/build.gradle

```diff
@@ -50,9 +51,29 @@ asciidoctor {
 
         into '.'
     }
+
+    ext.capturedOutput = [ ]
+    def listener = { ext.capturedOutput << it } as StandardOutputListener
+
+    logging.addStandardErrorListener(listener)
+    logging.addStandardOutputListener(listener)
+
+    doLast {
+        logging.removeStandardOutputListener(listener)
+        logging.removeStandardErrorListener(listener)
+        ext.capturedOutput.join('').with { output ->
+            if (output =~ /WARNING:/) {
+                throw new RuntimeException("See output above for specific cause of failure(s).")
+            }
+        }
+    }
+}
+
+linkchecker {
+    startFileName = "$buildDir/asciidoc/html5"
 }
 
-task('publishDocs', dependsOn: asciidoctor, group: 'release', description: 'Adds the new version of the docs to github pages') << {
+task('publishDocs', dependsOn: tasks.linkchecker, group: 'release', description: 'Adds the new version of the docs to github pages') << {
```

Comment:
I would leave this as it was, and make verify depend on the linkchecker

Reply:
So instead of:
* `build.dependsOn tasks.linkchecker`

make it:
* `verify.dependsOn tasks.linkchecker`

at line 137?

Reply:
Or would it be:
* `test.dependsOn tasks.linkchecker`

Reply:
yep, it better represents what part of the lifecycle of the build it is. And build already depends on verify. 
And the publish task depends on our overall gate.
https://github.com/rackerlabs/repose/blob/master/repose-aggregator/build.gradle#L116

Reply:
Gradle Java plugin tasks descriptions and lifecycle here:
https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_tasks

That has helped me a bunch. And, to me, it seems like you want:
`check.dependsOn tasks.linkchecker`

Is that what @adrianjgeorge meant? I don't think we have a `verify` task?

Reply:
You are correct that is what i meant. Where did i get verify? Maybe that's a maven goal?